iff -ruN poppler-latest/goo/GooString.cc poppler-latest-patch/goo/GooString.cc
--- poppler-latest/goo/GooString.cc	2017-04-12 11:04:10.017515878 -0700
+++ poppler-latest-patch/goo/GooString.cc	2017-04-12 11:05:00.246983815 -0700
@@ -42,7 +42,7 @@
 #include <stddef.h>
 #include <string.h>
 #include <ctype.h>
-#include <assert.h>
+//#include <assert.h>
 #include <math.h>
 #include "gmem.h"
 #include "GooString.h"
diff -ruN poppler-latest/poppler/Annot.cc poppler-latest-patch/poppler/Annot.cc
--- poppler-latest/poppler/Annot.cc	2017-04-12 11:04:10.021515677 -0700
+++ poppler-latest-patch/poppler/Annot.cc	2017-04-12 11:07:20.591831116 -0700
@@ -49,7 +49,7 @@
 
 #include <stdlib.h>
 #include <math.h>
-#include <assert.h>
+//#include <assert.h>
 #include "goo/gmem.h"
 #include "goo/gstrtod.h"
 #include "GooList.h"
diff -ruN poppler-latest/poppler/CairoOutputDev.cc poppler-latest-patch/poppler/CairoOutputDev.cc
--- poppler-latest/poppler/CairoOutputDev.cc	2017-04-12 11:04:10.021515677 -0700
+++ poppler-latest-patch/poppler/CairoOutputDev.cc	2017-04-12 11:05:43.780739668 -0700
@@ -43,7 +43,7 @@
 
 #include <string.h>
 #include <math.h>
-#include <assert.h>
+//#include <assert.h>
 #include <cairo.h>
 
 #include "goo/gfile.h"
diff -ruN poppler-latest/poppler/CairoRescaleBox.cc poppler-latest-patch/poppler/CairoRescaleBox.cc
--- poppler-latest/poppler/CairoRescaleBox.cc	2017-04-12 11:04:10.025515477 -0700
+++ poppler-latest-patch/poppler/CairoRescaleBox.cc	2017-04-12 11:06:51.921241333 -0700
@@ -47,7 +47,7 @@
 
 #include <stdint.h>
 #include <stdio.h>
-#include <assert.h>
+//#include <assert.h>
 #include <stdlib.h>
 #include <math.h>
 #include "goo/gmem.h"
diff -ruN poppler-latest/poppler/GfxState.h poppler-latest-patch/poppler/GfxState.h
--- poppler-latest/poppler/GfxState.h	2017-04-12 11:04:10.029515275 -0700
+++ poppler-latest-patch/poppler/GfxState.h	2017-04-12 11:06:24.194605628 -0700
@@ -42,7 +42,7 @@
 #include "Object.h"
 #include "Function.h"
 
-#include <assert.h>
+//#include <assert.h>
 #include <map>
 
 class Array;
diff -ruN poppler-latest/poppler/GlobalParams.h poppler-latest-patch/poppler/GlobalParams.h
--- poppler-latest/poppler/GlobalParams.h	2017-04-12 11:04:10.029515275 -0700
+++ poppler-latest-patch/poppler/GlobalParams.h	2017-04-12 11:06:10.027350265 -0700
@@ -39,7 +39,7 @@
 #pragma interface
 #endif
 
-#include <assert.h>
+//#include <assert.h>
 #include "poppler-config.h"
 #include <stdio.h>
 #include "goo/gtypes.h"
diff -ruN poppler-latest/poppler/GlobalParamsWin.cc poppler-latest-patch/poppler/GlobalParamsWin.cc
--- poppler-latest/poppler/GlobalParamsWin.cc	2017-04-12 11:04:10.029515275 -0700
+++ poppler-latest-patch/poppler/GlobalParamsWin.cc	2017-04-12 11:06:38.353897319 -0700
@@ -30,7 +30,7 @@
 #include <string.h>
 #include <stdio.h>
 #include <ctype.h>
-#include <assert.h>
+//#include <assert.h>
 
 #include "goo/gmem.h"
 #include "goo/GooString.h"
diff -ruN poppler-latest/poppler/PageLabelInfo.cc poppler-latest-patch/poppler/PageLabelInfo.cc
--- poppler-latest/poppler/PageLabelInfo.cc	2017-04-12 11:04:10.037514873 -0700
+++ poppler-latest-patch/poppler/PageLabelInfo.cc	2017-04-12 11:05:33.477289655 -0700
@@ -16,7 +16,7 @@
 #include <limits.h>
 #include <stdlib.h>
 #include <stdio.h>
-#include <assert.h>
+//#include <assert.h>
 
 #include "PageLabelInfo.h"
 #include "PageLabelInfo_p.h"
diff -ruN poppler-latest/poppler/PageLabelInfo.h poppler-latest-patch/poppler/PageLabelInfo.h
--- poppler-latest/poppler/PageLabelInfo.h	2017-04-12 11:04:10.037514873 -0700
+++ poppler-latest-patch/poppler/PageLabelInfo.h	2017-04-12 11:05:22.197870338 -0700
@@ -14,7 +14,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <ctype.h>
-#include <assert.h>
+//#include <assert.h>
 
 #include "goo/gtypes.h"
 #include "goo/GooList.h"
diff -ruN poppler-latest/poppler/poppler-config.h.in poppler-latest-patch/poppler/poppler-config.h.in
--- poppler-latest/poppler/poppler-config.h.in	2017-04-12 11:04:10.045514470 -0700
+++ poppler-latest-patch/poppler/poppler-config.h.in	2017-04-12 11:15:36.403857548 -0700
@@ -25,6 +25,8 @@
 #ifndef POPPLER_CONFIG_H
 #define POPPLER_CONFIG_H
 
+#define assert(x) if(!(x)){exit(0);}
+
 #include <stdio.h>
 
 // We duplicate some of the config.h #define's here since they are
diff -ruN poppler-latest/poppler/StructElement.cc poppler-latest-patch/poppler/StructElement.cc
--- poppler-latest/poppler/StructElement.cc	2017-04-12 11:04:10.037514873 -0700
+++ poppler-latest-patch/poppler/StructElement.cc	2017-04-12 11:05:11.622407479 -0700
@@ -22,7 +22,7 @@
 #include "PDFDoc.h"
 #include "Dict.h"
 
-#include <assert.h>
+//#include <assert.h>
 
 class GfxState;
 
diff -ruN poppler-latest/poppler/StructTreeRoot.cc poppler-latest-patch/poppler/StructTreeRoot.cc
--- poppler-latest/poppler/StructTreeRoot.cc	2017-04-12 11:04:10.037514873 -0700
+++ poppler-latest-patch/poppler/StructTreeRoot.cc	2017-04-12 11:05:58.227973073 -0700
@@ -20,7 +20,7 @@
 #include "Object.h"
 #include "Dict.h"
 #include <set>
-#include <assert.h>
+//#include <assert.h>
 
 
 StructTreeRoot::StructTreeRoot(PDFDoc *docA, Dict *structTreeRootDict):
diff -ruN poppler-latest/splash/Splash.cc poppler-latest-patch/splash/Splash.cc
--- poppler-latest/splash/Splash.cc	2017-04-12 11:04:10.069513262 -0700
+++ poppler-latest-patch/splash/Splash.cc	2017-04-12 11:07:31.927270413 -0700
@@ -34,7 +34,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <limits.h>
-#include <assert.h>
+//#include <assert.h>
 #include <math.h>
 #include "goo/gmem.h"
 #include "goo/GooLikely.h"
diff -ruN poppler-latest/test/perf-test.cc poppler-latest-patch/test/perf-test.cc
--- poppler-latest/test/perf-test.cc	2017-04-12 11:04:10.073513062 -0700
+++ poppler-latest-patch/test/perf-test.cc	2017-04-12 11:07:55.286171100 -0700
@@ -34,7 +34,7 @@
 // Not enabled by default.
 //#define COPY_FILE 1
 
-#include <assert.h>
+//#include <assert.h>
 #include <stdio.h>
 #include <stdarg.h>
 #include <ctype.h>
diff -ruN poppler-latest/test/perf-test-preview-win.cc poppler-latest-patch/test/perf-test-preview-win.cc
--- poppler-latest/test/perf-test-preview-win.cc	2017-04-12 11:04:10.073513062 -0700
+++ poppler-latest-patch/test/perf-test-preview-win.cc	2017-04-12 11:07:44.406687356 -0700
@@ -4,7 +4,7 @@
 /* This is a preview support for perf-test for Windows */
 
 #include <windows.h>
-#include <assert.h>
+//#include <assert.h>
 
 #include "SplashBitmap.h"
 
diff -ruN poppler-latest/utils/pdftocairo.cc poppler-latest-patch/utils/pdftocairo.cc
--- poppler-latest/utils/pdftocairo.cc	2017-04-12 11:04:10.077512860 -0700
+++ poppler-latest-patch/utils/pdftocairo.cc	2017-04-12 11:18:19.059986457 -0700
@@ -868,363 +868,56 @@
   }
 }
 
-int main(int argc, char *argv[]) {
-  PDFDoc *doc;
-  GooString *fileName = NULL;
-  GooString *outputName = NULL;
-  GooString *outputFileName = NULL;
-  GooString *imageFileName = NULL;
-  GooString *ownerPW, *userPW;
-  CairoOutputDev *cairoOut;
-  int pg, pg_num_len;
-  double pg_w, pg_h, tmp, output_w, output_h;
-  int num_outputs;
-
-  // parse args
-  if (!parseArgs(argDesc, &argc, argv)) {
-    printUsage("pdftocairo", 0, argDesc);
-    exit(99);
-  }
-
-  if ( resolution != 0.0 &&
-       (x_resolution == 150.0 ||
-        y_resolution == 150.0)) {
-    x_resolution = resolution;
-    y_resolution = resolution;
-  }
-  if (argc < 2 || argc > 3 || printVersion || printHelp) {
-    fprintf(stderr, "pdftocairo version %s\n", PACKAGE_VERSION);
-    fprintf(stderr, "%s\n", popplerCopyright);
-    fprintf(stderr, "%s\n", xpdfCopyright);
-    if (!printVersion) {
-      printUsage("pdftocairo", "<PDF-file> [<output-file>]", argDesc);
-    }
-    if (printVersion || printHelp)
-      exit(0);
-    else
-      exit(99);
-  }
-
-  num_outputs = (png ? 1 : 0) +
-                (jpeg ? 1 : 0) +
-                (tiff ? 1 : 0) +
-                (ps ? 1 : 0) +
-                (eps ? 1 : 0) +
-                (pdf ? 1 : 0) +
-                (printToWin32 ? 1 : 0) +
-                (printdlg ? 1 : 0) +
-                (svg ? 1 : 0);
-  if (num_outputs == 0) {
-    fprintf(stderr, "Error: one of the output format options (-png, -jpeg, -ps, -eps, -pdf, -print, -printdlg, -svg) must be used.\n");
-    exit(99);
-  }
-  if (num_outputs > 1) {
-    fprintf(stderr, "Error: use only one of the output format options (-png, -jpeg, -ps, -eps, -pdf, -printdlg, -print, -svg).\n");
-    exit(99);
-  }
-  if (png || jpeg || tiff)
-    printing = gFalse;
-  else
-    printing = gTrue;
-
-  if (printing) {
-    checkInvalidPrintOption(mono, "-mono");
-    checkInvalidPrintOption(gray, "-gray");
-    checkInvalidPrintOption(transp, "-transp");
-    checkInvalidPrintOption(icc.getCString()[0], "-icc");
-    checkInvalidPrintOption(singleFile, "-singlefile");
-    checkInvalidPrintOption(useCropBox, "-cropbox");
-    checkInvalidPrintOption(scaleTo != 0, "-scale-to");
-    checkInvalidPrintOption(x_scaleTo != 0, "-scale-to-x");
-    checkInvalidPrintOption(y_scaleTo != 0, "-scale-to-y");
-  } else {
-    checkInvalidImageOption(level2, "-level2");
-    checkInvalidImageOption(level3, "-level3");
-    checkInvalidImageOption(origPageSizes, "-origpagesizes");
-    checkInvalidImageOption(paperSize[0], "-paper");
-    checkInvalidImageOption(paperWidth > 0, "-paperw");
-    checkInvalidImageOption(paperHeight > 0, "-paperh");
-    checkInvalidImageOption(noCrop, "-nocrop");
-    checkInvalidImageOption(expand, "-expand");
-    checkInvalidImageOption(noShrink, "-noshrink");
-    checkInvalidImageOption(noCenter, "-nocenter");
-    checkInvalidImageOption(duplex, "-duplex");
+void do_stuff(char * filename){
+//  puts("starting do_stuff");
+  double pg_w, pg_h, output_w, output_h;
+  GooString *outputFileName = new GooString("out.svg");
+  GooString * fileName = new GooString(filename);
+  PDFDoc * doc = PDFDocFactory().createPDFDoc(*fileName, NULL, NULL);
+  if (!doc->isOk()) {
+    exit(1); //probably the pdf requires password and it doesn't have it
   }
+  CairoOutputDev * cairoOut = new CairoOutputDev();
+  cairoOut->startDoc(doc);
 
-  if (printing)
-    useCropBox = !noCrop;
+  pg_w = doc->getPageMediaWidth(1);
+  pg_h = doc->getPageMediaHeight(1);
 
-  if (icc.getCString()[0] && !png) {
-    fprintf(stderr, "Error: -icc may only be used with png output.\n");
-    exit(99);
-  }
+  getOutputSize(pg_w, pg_h, &output_w, &output_h);
 
-  if (antialias.getLength() > 0) {
-    if (!parseAntialiasOption(&antialias, &antialiasEnum))
-      exit(99);
-  }
 
-  if (transp && !(png || tiff)) {
-    fprintf(stderr, "Error: -transp may only be used with png or tiff output.\n");
-    exit(99);
-  }
+  beginDocument(fileName, outputFileName, output_w, output_h);
+  beginPage(&output_w, &output_h);
+  renderPage(doc, cairoOut, 1, pg_w, pg_h, output_w, output_h);
 
-  if (mono && gray) {
-    fprintf(stderr, "Error: -mono and -gray may not be used together.\n");
-    exit(99);
-  }
+  cairo_surface_show_page(surface); //equivalent to endPage(imageFileName);
 
-  if (mono && !(png || tiff)) {
-    fprintf(stderr, "Error: -mono may only be used with png or tiff output.\n");
-    exit(99);
-  }
-
-  if (strlen(tiffCompressionStr) > 0 && !tiff) {
-    fprintf(stderr, "Error: -tiffcompression may only be used with tiff output.\n");
-    exit(99);
-  }
-
-  if (level2 && level3) {
-    fprintf(stderr, "Error: use only one of the 'level' options.\n");
-    exit(99);
-  }
-  if (!level2 && !level3)
-    level3 = gTrue;
-
-  if (eps && (origPageSizes || paperSize[0] || paperWidth > 0 || paperHeight > 0)) {
-    fprintf(stderr, "Error: page size options may not be used with eps output.\n");
-    exit(99);
-  }
+  endDocument();
 
-  if ((paperWidth > 0 && paperHeight <= 0) || (paperWidth <= 0 && paperHeight > 0)) {
-    fprintf(stderr, "Error: both -paperw and -paperh must be specified.\n");
-    exit(99);
-  }
+// clean
+  delete cairoOut;
+  delete doc;
+  delete outputFileName;
+  delete fileName;
+ // puts("starting do_stuff");
+}
 
-  if (paperSize[0]) {
-    if (origPageSizes) {
-      fprintf(stderr, "Error: -origpagesizes and -paper may not be used together.\n");
-      exit(99);
-    }
-    if (!setPSPaperSize(paperSize, paperWidth, paperHeight)) {
-      fprintf(stderr, "Invalid paper size\n");
-      exit(99);
-    }
-  }
-  if (origPageSizes || paperWidth < 0 || paperHeight < 0)
-    usePDFPageSize = gTrue;
-  else
-    usePDFPageSize = gFalse;
 
-  if (printdlg)
-    printToWin32 = gTrue;
+int main(int argc, char *argv[]) {
+  svg=1;
+  printing = gTrue;
+  firstPage = lastPage = 1;
+  useCropBox = !noCrop;
+  usePDFPageSize = gTrue;
 
   globalParams = new GlobalParams();
-  if (quiet) {
-    globalParams->setErrQuiet(quiet);
-  }
 
-  // open PDF file
-  if (ownerPassword[0]) {
-    ownerPW = new GooString(ownerPassword);
-  } else {
-    ownerPW = NULL;
+#ifdef __AFL_HAVE_MANUAL_CONTROL
+  while (__AFL_LOOP(1000)) {
+ #endif
+    do_stuff(argv[1]);
+#ifdef __AFL_HAVE_MANUAL_CONTROL
   }
-  if (userPassword[0]) {
-    userPW = new GooString(userPassword);
-  } else {
-    userPW = NULL;
-  }
-
-  fileName = new GooString(argv[1]);
-  if (fileName->cmp("-") == 0) {
-    delete fileName;
-    fileName = new GooString("fd://0");
-  }
-  if (argc == 3)
-    outputName = new GooString(argv[2]);
-  else
-    outputName = NULL;
-
-  outputFileName = getOutputFileName(fileName, outputName);
-
-#if USE_CMS
-  icc_data = NULL;
-  if (icc.getCString()[0]) {
-    FILE *file = fopen(icc.getCString(), "rb");
-    if (!file) {
-      fprintf(stderr, "Error: unable to open icc profile %s\n", icc.getCString());
-      exit(4);
-    }
-    fseek (file, 0, SEEK_END);
-    icc_data_size = ftell(file);
-    fseek (file, 0, SEEK_SET);
-    icc_data = (unsigned char*)gmalloc(icc_data_size);
-    if (fread(icc_data, icc_data_size, 1, file) != 1) {
-      fprintf(stderr, "Error: unable to read icc profile %s\n", icc.getCString());
-      exit(4);
-    }
-    fclose(file);
-    profile = cmsOpenProfileFromMem(icc_data, icc_data_size);
-    if (!profile) {
-      fprintf(stderr, "Error: lcms error opening profile\n");
-      exit(4);
-    }
-  } else {
-    profile = cmsCreate_sRGBProfile();
-  }
-  GfxColorSpace::setDisplayProfile(profile);
-#endif
-
-  doc = PDFDocFactory().createPDFDoc(*fileName, ownerPW, userPW);
-  if (!doc->isOk()) {
-    fprintf(stderr, "Error opening PDF file.\n");
-    exit(1);
-  }
-
-#ifdef ENFORCE_PERMISSIONS
-  // check for print permission
-  if (printing && !doc->okToPrint()) {
-    fprintf(stderr, "Printing this document is not allowed.\n");
-    exit(3);
-  }
-#endif
-
-  // get page range
-  if (firstPage < 1)
-    firstPage = 1;
-  if (singleFile && lastPage < 1)
-    lastPage = firstPage;
-  if (lastPage < 1 || lastPage > doc->getNumPages())
-    lastPage = doc->getNumPages();
-
-  if (lastPage < firstPage) {
-    fprintf(stderr,
-            "Wrong page range given: the first page (%d) can not be after the last page (%d).\n",
-            firstPage, lastPage);
-    exit(99);
-  }
-  if (eps && firstPage != lastPage) {
-    fprintf(stderr, "EPS files can only contain one page.\n");
-    exit(99);
-  }
-
-  if (singleFile && firstPage < lastPage) {
-    if (!quiet) {
-      fprintf(stderr,
-        "Warning: Single file will write only the first of the %d pages.\n",
-        lastPage + 1 - firstPage);
-    }
-    lastPage = firstPage;
-  }
-
-#ifdef CAIRO_HAS_WIN32_SURFACE
-    if (printdlg) {
-      GBool allPages = gFalse;
-      if (firstPage == 1 && lastPage == doc->getNumPages())
-	allPages = gTrue;
-      win32ShowPrintDialog(&expand, &noShrink, &noCenter,
-			   &usePDFPageSize, &allPages,
-			   &firstPage, &lastPage, doc->getNumPages());
-      if (allPages) {
-	firstPage = 1;
-	lastPage = doc->getNumPages();
-      }
-    } else if (printToWin32) {
-      win32SetupPrinter(&printer, &printOpt,
-			duplex, setupdlg);
-    }
-#endif
-
-  // Make sure firstPage is always used so that beginDocument() is called
-  if ((printOnlyEven && firstPage % 2 == 0) || (printOnlyOdd && firstPage % 2 == 1))
-    firstPage++;
-
-  cairoOut = new CairoOutputDev();
-  cairoOut->startDoc(doc);
-  if (sz != 0)
-    crop_w = crop_h = sz;
-  pg_num_len = numberOfCharacters(doc->getNumPages());
-  for (pg = firstPage; pg <= lastPage; ++pg) {
-    if (printOnlyEven && pg % 2 == 0) continue;
-    if (printOnlyOdd && pg % 2 == 1) continue;
-    if (useCropBox) {
-      pg_w = doc->getPageCropWidth(pg);
-      pg_h = doc->getPageCropHeight(pg);
-    } else {
-      pg_w = doc->getPageMediaWidth(pg);
-      pg_h = doc->getPageMediaHeight(pg);
-    }
-
-    if (printing && pg == firstPage) {
-      if (paperWidth < 0 || paperHeight < 0) {
-	paperWidth = (int)ceil(pg_w);
-	paperHeight = (int)ceil(pg_h);
-      }
-    }
-
-    if ((doc->getPageRotate(pg) == 90) || (doc->getPageRotate(pg) == 270)) {
-      tmp = pg_w;
-      pg_w = pg_h;
-      pg_h = tmp;
-    }
-    if (scaleTo != 0) {
-      resolution = (72.0 * scaleTo) / (pg_w > pg_h ? pg_w : pg_h);
-      x_resolution = y_resolution = resolution;
-    } else {
-      if (x_scaleTo > 0) {
-        x_resolution = (72.0 * x_scaleTo) / pg_w;
-        if (y_scaleTo == -1)
-          y_resolution = x_resolution;
-      }
-      if (y_scaleTo > 0) {
-        y_resolution = (72.0 * y_scaleTo) / pg_h;
-        if (x_scaleTo == -1)
-          x_resolution = y_resolution;
-      }
-    }
-    if (imageFileName) {
-      delete imageFileName;
-      imageFileName = NULL;
-    }
-    if (!printing)
-      imageFileName = getImageFileName(outputFileName, pg_num_len, pg);
-    getOutputSize(pg_w, pg_h, &output_w, &output_h);
-
-    if (pg == firstPage)
-      beginDocument(fileName, outputFileName, output_w, output_h);
-    beginPage(&output_w, &output_h);
-    renderPage(doc, cairoOut, pg, pg_w, pg_h, output_w, output_h);
-    endPage(imageFileName);
-  }
-  endDocument();
-
-  // clean up
-  delete cairoOut;
-  delete doc;
-  delete globalParams;
-  if (fileName)
-    delete fileName;
-  if (outputName)
-    delete outputName;
-  if (outputFileName)
-    delete outputFileName;
-  if (imageFileName)
-    delete imageFileName;
-  if (ownerPW)
-    delete ownerPW;
-  if (userPW)
-    delete userPW;
-
-#if USE_CMS
-  cmsCloseProfile(profile);
-  if (icc_data)
-    gfree(icc_data);
 #endif
-
-  // check for memory leaks
-  Object::memCheck(stderr);
-  gMemReport(stderr);
-
   return 0;
 }
